src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -2296,8 +2296,7 @@ void MarkTypeForDynamicallyAccessedMembers (ref ReflectionPatternContext reflect
 					MarkField (ref reflectionContext, field, dependencyKind);
 					break;
 				case TypeDefinition nestedType:
-					DependencyInfo nestedDependencyInfo = new DependencyInfo (dependencyKind, reflectionContext.Source);
-					reflectionContext.RecordRecognizedPattern (nestedType, () => _markStep.MarkEntireType (nestedType, includeBaseAndInterfaceTypes: true, nestedDependencyInfo));
+					MarkType (ref reflectionContext, nestedType, dependencyKind);
```

Comment:
Can you add a test case along these lines. The base type one would look like:

```C#
class Base<[DAM(All)] T> {}

class TestType : Base<TestType> {}
```

Similarly for interfaces.

Another one is attributes on members: We already have an issue with this on properties and other members - see https://github.com/mono/linker/issues/2196 (I just created this). The All is basically just another case of the same problem for these members.

Reply:
Yup, the first example is already covered by this test: https://github.com/mono/linker/blob/e21bb44ca909d33c18acf3ed0b68f94a413351a5/test/Mono.Linker.Tests.Cases/DataFlow/AttributeConstructorDataflow.cs#L96-L100.

I added the second testcase you suggested - I think generic type parameters shouldn't be an issue since the DAM binder only gives back members and doesn't look at the instantiated base generic type at all (instead it relies on normal MarkType logic to mark generic arguments).

Good find about attributes on properties/events.

Reply:
>  (instead it relies on normal MarkType logic to mark generic arguments).

That was the point - I don't think `MarkType` was written with reentrancy in mind - it seems to handle it, which is good, but I thought it would be good to have tests for it. Before data flow, there were only a few ways to get to the reentrancy, now there are many more.

But I'm fine if we try to cover this later on.

Reply:
Yeah, it looks like it was at least given some thought but it's good to have more tests as we add more ways to get to MarkType. I did add the generic tests you suggested above so I think that is covered.